catkin_virtualenv/cmake/catkin_generate_virtualenv.cmake

```diff
@@ -32,19 +32,64 @@ function(catkin_generate_virtualenv)
     set(ARG_PYTHON_INTERPRETER "python3")
   endif()
 
-  if(NOT DEFINED ARG_USE_SYSTEM_PACKAGES OR ARG_USE_SYSTEM_PACKAGES)
-    message(STATUS "Using system site packages")
-    set(venv_args "--use-system-packages")
+  
+
+  if(DEFINED ARG_USE_UV)
+    message(STATUS "USING UV")
+    set(use_uv "--use-uv")
+
+    if( ARG_ISOLATE_REQUIREMENTS )
+      message( FATAL_ERROR "You can use ISOLATE_REQUIREMENTS in conjunction with USE_UV" )
```

Comment:
Did you mean `You can't use ISOLATE_REQUIREMENTS...`

Reply:
Depends on what that means. Do you not want to use `uv` at all, do you want your own cache (why?) 
I'll have to look at what `ISOLATE_REQUIREMENTS` means a bit more, for some reason I thought it had to do with venv chaining and not necessarily using `system-packages`

Reply:
`ISOLATE_REQUIREMENTS` does indeed cause a package not to inherit its dependencies `requirements.txt`. Interestingly enough we don't seem to use it anywhere anymore, and I can't remember why we added it.